src/Nox.Generator/Application/Factories/EntityMapper.template.cs

```diff
@@ -46,29 +46,45 @@ public override void MapToEntity({{entity.Name}} entity, Entity entityDefinition
     {{- #to be removed when we support all types -}}
     {{- if attribute.Type == "Date" || attribute.Type == "CurrencyNumber" || attribute.Type == "InternetDomain" || attribute.Type == "Json" || attribute.Type == "JwtToken" || attribute.Type == "DateTimeDuration" || attribute.Type == "DateTimeSchedule" || attribute.Type == "DayOfWeek" || attribute.Type == "Color" || attribute.Type == "Year" || attribute.Type == "DateTime" || attribute.Type == "Html" || attribute.Type == "Guid" || attribute.Type == "CurrencyCode3" ||  attribute.Type == "CultureCode"|| attribute.Type == "IpAddress" || attribute.Type == "Markdown" ||  attribute.Type == "CountryNumber" || attribute.Type == "Distance" || attribute.Type == "Boolean" || attribute.Type == "LatLong" || attribute.Type == "TranslatedText" || attribute.Type == "File" || attribute.Type == "VatNumber" || attribute.Type == "StreetAddress" || attribute.Type == "Money" || attribute.Type == "Text" || attribute.Type == "Number" || attribute.Type == "Length" || attribute.Type == "Area"  || attribute.Type == "Email" || attribute.Type == "TimeZoneCode" || attribute.Type == "MacAddress" || attribute.Type == "Uri" || attribute.Type == "Url" || attribute.Type == "User" || attribute.Type == "Volume" || attribute.Type == "Weight" || attribute.Type == "Yaml" || attribute.Type == "PhoneNumber" || attribute.Type == "Temperature" || attribute.Type == "Percentage" || attribute.Type == "CountryCode2"  || attribute.Type == "Month" }}
-        noxTypeValue = CreateNoxType<Nox.Types.{{attribute.Type}}>(entityDefinition,"{{attribute.Name}}",dto.{{attribute.Name}});
-        if(noxTypeValue != null)
+        noxTypeValue = CreateNoxType<Nox.Types.{{attribute.Type}}>(entityDefinition, "{{attribute.Name}}", dto.{{attribute.Name}});
+        if (noxTypeValue != null)
         {        
             entity.{{attribute.Name}} = noxTypeValue;
         }
     {{- else }}
 
         // TODO map {{attribute.Name}} {{attribute.Type}} remaining types and remove if else
-    {{- end}}        
+    {{- end }}
+    {{- end }}
+    {{ for relationship in entity.Relationships }}
+    {{- if (relationship.Relationship == "ZeroOrOne" || relationship.Relationship == "ExactlyOne") && relationship.ShouldGenerateForeignOnThisSide}}
+
+        /// <summary>
+        /// {{entity.Name}} {{relationship.Description}} {{relationship.Relationship}} {{relationship.EntityPlural}}
+        /// </summary>
+        noxTypeValue = CreateNoxType<Nox.Types.{{relationship.Related.Entity.Keys[0].Type}}>(entityDefinition, "{{relationship.Name}}", dto.{{relationship.Entity}}Id);
+        if (noxTypeValue != null)
+        {        
+            entity.{{relationship.Entity}}Id = noxTypeValue;
+        }
+    {{-end}}
     {{- end }}
     }
 
     public override void PartialMapToEntity({{entity.Name}} entity, Entity entityDefinition, Dictionary<string, dynamic> updatedProperties)
     {
+#pragma warning disable CS0168 // Variable is assigned but its value is never used
```

Comment:
we need, because if we don't, we have to declare it in out parameter, and if so, we will get "Error	CS0136	A local or parameter named 'value' cannot be declared in this scope because that name is used in an enclosing local scope to define a local or parameter"